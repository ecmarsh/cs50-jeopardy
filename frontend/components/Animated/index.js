import Ease from './_Ease';
import Fade from './_Fade';
import Grow from './_Grow';
import Loud from './_Loud';
import Shift from './_Shift';
import Slide from './_Slide';
import Tally from './_Tally';
import Tree from './_Tree';
import Unfocus from './_Unfocus';

export { Ease, Fade, Grow, Loud, Shift, Slide, Tally, Tree, Unfocus };
