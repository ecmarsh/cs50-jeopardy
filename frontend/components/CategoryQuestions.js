import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import AddToCategory from './CategoryQuestionAdd';
import { categoriesPerPage } from '../config';

const CATEGORY_QUESTIONS_QUERY = gql`
  query CATEGORY_QUESTIONS_QUERY($categoryId: ID) {
    categoryQuestions(where: { category: { id: $categoryId } }) {
      id
      difficulty
      isDouble
      answered
      question {
        id
        question
        answer
      }
      category {
        id
        name
      }
    }
  }
`;

const CategoryQuestions = ({ gameName, categoryId, round }) => (
  <Query query={CATEGORY_QUESTIONS_QUERY} variables={{ categoryId }}>
    {({ data, error }) => {
      if (error) return <p>Error...</p>;
      if (data.categoryQuestions)
        return (
          <>
            {data.categoryQuestions.map((categoryQuestion, index) => (
              <AddToCategory
                key={categoryQuestion.id}
                gameName={gameName}
                categoryId={categoryId}
                categoryName={categoryQuestion.category.name}
                categoryQuestionId={categoryQuestion.id}
                difficulty={categoryQuestion.difficulty}
                question={categoryQuestion.question}
                loaded={categoryQuestion.question !== null}
                skip={categoriesPerPage * (round - 1)}
                cacheCategoryIndex={index}
              />
            ))}
          </>
        );
    }}
  </Query>
);

export default CategoryQuestions;
export { CATEGORY_QUESTIONS_QUERY };
