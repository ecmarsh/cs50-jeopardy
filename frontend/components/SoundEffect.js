import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

const LOCAL_STATE_QUERY = gql`
  query {
    soundOn @client
  }
`;

const sounds = [
  { name: 'Aww', file: 'aww', duration: 2000 },
  { name: 'Clap', file: 'applause', duration: 7000 },
  { name: 'Correct', file: 'correct', duration: 2000 },
  { name: 'Daily Double', file: 'daily-double', duration: 3000 },
  { name: 'Fill Board', file: 'fill-board', duration: 4000 },
  { name: 'Final Jeopardy', file: 'final-jeopardy', duration: 32000 },
  { name: 'Thinking', file: 'thinking', duration: 61000 },
  { name: 'Times Up', file: 'times-up', duration: 1000 },
  { name: 'Wrong', file: 'wrong', duration: 550 },
];

const SoundEffect = ({ sound, disabled, ...rest }) => (
  <Query query={LOCAL_STATE_QUERY}>
    {({ data: { soundOn } }) => {
      return (
        <span>
          {(soundOn || disabled === false) && (
            <audio autoPlay src={`../static/sounds/${sound}.mp3`} {...rest} />
          )}
        </span>
      );
    }}
  </Query>
);
SoundEffect.propTypes = {
  sound: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
};
SoundEffect.defaultProps = {
  sound: 'wrong',
};

export default SoundEffect;
export { LOCAL_STATE_QUERY, sounds };
