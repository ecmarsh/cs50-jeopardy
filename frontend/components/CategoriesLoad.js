import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { categoriesPerPage } from '../config';
import CategoryQuestions from './CategoryQuestions';
import { Heading } from './Views';
import Pagination from './Pagination';
import Error from './ErrorMessage';
import Main from './styles/Main';
import { Divider } from './styles/Utilities';

const CATEGORIES_QUERY = gql`
  query CATEGORIES_QUERY($gameName: String, $first: Int, $skip: Int) {
    categories(
      where: { game: { name: $gameName } }
      first: $first
      skip: $skip
    ) {
      id
      name
    }
  }
`;

class LoadCategories extends Component {
  state = {};

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { game, round, headingProps } = this.props;
    return (
      <Query
        query={CATEGORIES_QUERY}
        variables={{
          gameName: game,
          first: categoriesPerPage,
          skip: categoriesPerPage * (round - 1),
        }}
      >
        {({ data, loading, error }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <Error error={error} />;
          if (!data.categories) {
            return <Heading game={game} {...headingProps} />;
          }
          return (
            <Main>
              <Heading game={game} {...headingProps} />
              <Divider style={{ margin: '0 auto' }} />
              <Pagination gameName={game} round={round} />
              <div className="canvas">
                {data.categories.map(category => (
                  <div
                    key={'category' + category.id}
                    className="column-wrapper"
                  >
                    <ul>
                      <li>{category.name}</li>
                      <CategoryQuestions
                        categoryId={category.id}
                        gameName={game}
                        round={round}
                      />
                    </ul>
                  </div>
                ))}
              </div>
            </Main>
          );
        }}
      </Query>
    );
  }
}

export default LoadCategories;
export { CATEGORIES_QUERY };
