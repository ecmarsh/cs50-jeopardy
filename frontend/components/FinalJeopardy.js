import React, { useState, Fragment } from 'react';
import Router from 'next/router';
import { Query } from 'react-apollo';
import Error from './ErrorMessage';
import GameReset from './GameReset';
import { CATEGORY_QUESTIONS_QUERY } from './CategoryQuestions';
import formatScore from '../lib/formatScore';
import WagerInputs from './styles/Wagers';
import Cover from './styles/Cover';
import Button from './styles/Button';
import { TextEmph, Divider } from './styles/Utilities';
import { UpdateTeamScoreButton } from './ScoreTeams';
import RoundTimer from './Timer';
import { pluck } from 'ramda';
import SoundEffect from './SoundEffect';
import { Fade, Shift, Unfocus } from './Animated';

function FinalJeopardy({
  teams,
  gameName,
  config,
  category,
  toggleChosenTeam,
}) {
  const [wagers, setWagers] = useState([0, 0]);
  const [stage, setStage] = useState(0);
  const [qId, setQId] = useState('');
  const [question, setQuestion] = useState('');
  const [answer, setAnswer] = useState('');
  const [isAnswerShown, setIsAnswerShown] = useState(false);
  const nextStage = (override = false) =>
    setStage(prevStage => (prevStage += 1));
  const showAnswer = () => setIsAnswerShown(true);

  const winner = () => {
    if (teams[0].score < teams[1].score) {
      return `Team ${teams[1].name}`;
    } else if (teams[0].score > teams[1].score) {
      return `Team ${teams[0].name}`;
    }
    return `Both teams win!`; // Must be a tie
  };
  const overrides = {
    position: 'relative',
    top: '5px',
    width: '100%',
    height: '100%',
    transform: 'translateX(-50%)',
  };
  return (
    <Cover style={overrides}>
      <div className="fullscreen__inner">
        <div className="titles">
          <div className="titles__inner">
            <h3>{stage === 4 ? 'CS50 Jeopardy' : category.name}</h3>
          </div>
          <Divider />
        </div>
        {stage === 0 && (
          <TeamWagers
            teams={teams}
            nextStage={nextStage}
            setWagers={setWagers}
          />
        )}
        {stage === 1 && (
          <FinalJeopardyQuestion
            config={config}
            gameName={gameName}
            categoryId={category.id}
            nextStage={nextStage}
            liftQuestion={setQuestion}
            liftAnswer={setAnswer}
            liftQId={setQId}
          />
        )}
        {(stage === 2 || stage === 3) && (
          <PromptAnswer
            key={`promptAnswer${stage}`}
            teams={teams}
            index={stage - 2}
            wagers={wagers}
            nextStage={nextStage}
            question={question}
            answer={answer}
            isAnswerShown={isAnswerShown}
            showAnswer={showAnswer}
            qId={qId}
            categoryId={category.id}
            toggleChosenTeam={toggleChosenTeam}
            gameName={gameName}
          />
        )}
        {stage === 4 && (
          <EndOfGame
            winner={winner()}
            gameName={gameName}
            hasDoubleJeopardy={config.hasDoubleJeopardy}
          />
        )}
      </div>
    </Cover>
  );
}

function TeamWagers({ teams, setWagers, nextStage }) {
  const [wager0, setWager0] = useState(0);
  const [wager1, setWager1] = useState(0);
  const [step, changeStep] = useState(100);
  const [errMessage, setErrMessage] = useState('');
  const wagers = [wager0 | 0, wager1 | 0];

  const validate = () => {
    const scores = pluck('score', teams);
    let e = '';
    scores.forEach((score, i) => {
      if (wagers[i] < 0) e = 'Wagers must be positive!';
      if (score > 1000 && wagers[i] > score) e = 'Max wager is your score!';
      if (score < 1000 && wagers[i] > 1000)
        e = 'Max wager is $1,000 or your score for this game!';
    });
    setErrMessage(e);

    if (e) {
      return false;
    }
    return true;
  };

  const handleSubmit = () => {
    const validateInputs = validate();
    if (!validateInputs) return;

    setWagers(wagers);
    nextStage();
  };
  return (
    <Fragment>
      <div className="main__display">
        <WagerInputs>
          <TextEmph>Choose Final Wagers</TextEmph>
          {errMessage && (
            <p style={{ fontSize: 'smaller', opacity: '.8', color: 'red' }}>
              {errMessage}
            </p>
          )}

          {teams.map((team, i) => (
            <label key={`wager-${i}`} htmlFor={`wager${i}`}>
              {team.name}
              <span>
                ${' '}
                <input
                  type="number"
                  required
                  max={team.score <= 0 ? 1000 : team.score}
                  min={0}
                  value={wagers[i]}
                  step={step}
                  name={`wager${i}`}
                  onKeyUp={() => changeStep(1)}
                  onChange={e => {
                    if (!i) {
                      setWager0(e.target.value);
                    } else {
                      setWager1(e.target.value);
                    }
                  }}
                />
              </span>
            </label>
          ))}
          <p className="disclaimer">
            You may have to write down your team{"'"}s answer
          </p>
          <p className="disclaimer smaller">
            ...And cover your eyes for the other teams wager ðŸ™ˆ
          </p>
        </WagerInputs>
      </div>
      <div className="controls">
        <Divider />
        <Button game onClick={() => handleSubmit()}>
          Start Final Round
        </Button>
      </div>
    </Fragment>
  );
}

function FinalJeopardyQuestion({
  config,
  gameName,
  categoryId,
  nextStage,
  liftQId,
  liftQuestion,
  liftAnswer,
}) {
  return (
    <Query
      query={CATEGORY_QUESTIONS_QUERY}
      variables={{ categoryId: categoryId }}
    >
      {({ data: { categoryQuestions }, loading, error }) => {
        if (loading)
          return (
            <div className="main__display">
              <p className="question">Loading...</p>
            </div>
          );
        if (error) return <Error error={error} />;
        const { id, question } = categoryQuestions[0];
        if (config.hasRoundTimer) {
          liftQId(id);
          liftQuestion(question.question);
          liftAnswer(question.answer);
        }
        return (
          <Fragment>
            <div className="main__display">
              <p className="question">{question.question}</p>
              {config.hasRoundTimer && (
                <RoundTimer
                  key={`timerfinal`}
                  gameName={gameName}
                  round={3}
                  roundTime={config.roundTime}
                  finalTime={config.finalTime}
                  nextStage={nextStage}
                />
              )}
            </div>
            <div className="controls">
              <Divider />
              <Button
                game
                onClick={() => {
                  liftQId(id);
                  liftQuestion(question.question);
                  liftAnswer(question.answer);
                  nextStage();
                }}
              >
                End Round
              </Button>
            </div>
            <SoundEffect sound="final-jeopardy" loop />
          </Fragment>
        );
      }}
    </Query>
  );
}

function PromptAnswer({
  teams,
  wagers,
  index,
  nextStage,
  question,
  answer,
  isAnswerShown,
  showAnswer,
  categoryId,
  qId,
  toggleChosenTeam,
  gameName,
}) {
  return (
    <Fragment>
      <div
        className="main__display"
        style={{ alignSelf: 'start', paddingTop: '3rem' }}
      >
        {index === 0 ? (
          <Unfocus off={true}>{question}</Unfocus>
        ) : (
          <p className="question--unfocused question--mounted">{question}</p>
        )}
        {isAnswerShown && index === 0 && <Shift cName="answer">{answer}</Shift>}
        {isAnswerShown && index === 1 && <p className="answer">{answer}</p>}
        <Divider style={{ width: '50%', marginBottom: '3rem' }} />
        <Shift cName="other">
          {teams[index].name} wagered{' '}
          <span>{formatScore(Math.abs(wagers[index]))}</span>
        </Shift>
        <Shift cName="prompt">
          Please show your answer, Team {teams[index].name}.
        </Shift>
      </div>
      <div className="controls">
        <Divider />
        <div className="controls__ctas">
          {!isAnswerShown && (
            <Button game onClick={showAnswer}>
              Show Answer
            </Button>
          )}
          {isAnswerShown &&
            [1, -1].map((multiplier, i) => (
              <UpdateTeamScoreButton
                key={`rightWrong${multiplier}`}
                dataTest={'cta-button-' + i}
                gameName={gameName}
                chosenTeam={teams[index]}
                sumScoreWith={wagers[index] * multiplier}
                categoryId={categoryId}
                categoryQuestionId={qId}
                toggleView={nextStage}
                toggleChosenTeam={toggleChosenTeam}
              >
                {multiplier === 1 ? 'Correct' : 'Incorrect'}
              </UpdateTeamScoreButton>
            ))}
        </div>
      </div>
    </Fragment>
  );
}

function EndOfGame({ winner, gameName, hasDoubleJeopardy }) {
  return (
    <Fragment>
      <div className="main__display">
        <Fade cName="congrats">
          Congratulations,
          <br /> {winner}
        </Fade>
      </div>
      <div className="controls">
        <Divider />
        <GameReset gameName={gameName} hasDoubleJeopardy={hasDoubleJeopardy}>
          {resetGameMutation => (
            <Button
              game
              onClick={async e => {
                if (
                  confirm(`Are you sure you want to reset game ${gameName}`)
                ) {
                  e.preventDefault();
                  await resetGameMutation();
                  Router.push({
                    pathname: `/play`,
                    query: { game: gameName },
                  });
                }
              }}
            >
              Reset Game
            </Button>
          )}
        </GameReset>
      </div>
      <SoundEffect sound="applause" />
    </Fragment>
  );
}

export default FinalJeopardy;
