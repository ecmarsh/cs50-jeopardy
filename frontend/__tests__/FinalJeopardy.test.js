import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import FinalJeopardy from '../components/FinalJeopardy';
import { CATEGORY_QUESTIONS_QUERY } from '../components/CategoryQuestions';
import {
  fakeGame,
  fakeGameConfig,
  fakeCategoryQuestion,
} from '../lib/testUtils';
import { LOCAL_STATE_QUERY } from '../components/SoundEffect';

const game = fakeGame();
const config = fakeGameConfig();
const categoryQuestion = fakeCategoryQuestion();

const mockProps = {
  teams: game.teams,
  gameName: game.name,
  config,
  category: categoryQuestion.category,
  toggleChosenTeam: jest.fn(),
};

const mocks = [
  {
    request: {
      query: CATEGORY_QUESTIONS_QUERY,
      variables: { categoryId: categoryQuestion.category.id },
    },
    result: {
      data: { categoryQuestions: [categoryQuestion] },
    },
  },
  {
    request: {
      query: LOCAL_STATE_QUERY,
    },
    result: {
      data: { soundOn: false },
    },
  },
];

describe('<FinalJeopardy />', () => {
  it('renders final jeopardy form and matches snapshot', () => {
    const wrapper = mount(<FinalJeopardy {...mockProps} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('shows accurate control for each stage', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <FinalJeopardy {...mockProps} />
      </MockedProvider>
    );
    wrapper.find('button').simulate('click');
    await wait();
    wrapper.update();
    expect(wrapper.find('button').text()).toContain('End Round');
    wrapper.find('button').simulate('click');
    await wait();
    wrapper.update();
    expect(wrapper.find('button').text()).toContain('Show Answer');
    wrapper.find('button').simulate('click');
    await wait();
    wrapper.update();
    expect(wrapper.exists('p.answer')).toEqual(true);
    expect(wrapper.find('button[data-test="cta-button-0"]').text()).toContain(
      'Correct'
    );
  });

  it('renders countdown when enabled and matches snap', async () => {
    config.hasRoundTimer = true; // turn on timer
    config.finalTime = 30; // give it some seconds
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <FinalJeopardy {...mockProps} />
      </MockedProvider>
    );
    // next screen
    wrapper.find('button').simulate('click');
    await wait();
    wrapper.update();
    expect(wrapper.exists('RoundTimer')).toEqual(true);
    expect(toJSON(wrapper.find('RoundTimer'))).toMatchSnapshot();
  });
});
