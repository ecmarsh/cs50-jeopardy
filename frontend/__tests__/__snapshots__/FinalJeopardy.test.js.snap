// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinalJeopardy /> renders countdown when enabled and matches snap 1`] = `
<RoundTimer
  finalTime={30}
  gameName="Test Game"
  key="timerfinal"
  nextStage={[Function]}
  round={3}
  roundTime={0}
>
  <Timer__StyledTimer
    final={true}
  >
    <StyledComponent
      final={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Timer__StyledTimer-blbssp-0",
            "isStatic": false,
            "lastClassName": "cmzUqk",
            "rules": Array [
              "color:",
              [Function],
              ";position:fixed;bottom:1.5rem;left:50%;transform:translateX(-50%);",
              [Function],
              " span{color:",
              [Function],
              ";margin-left:1rem;font-size:1.1em;}",
            ],
          },
          "displayName": "Timer__StyledTimer",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Timer__StyledTimer-blbssp-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Timer__StyledTimer-blbssp-0 cmzUqk"
      >
        <Countdown
          animationTime={30}
          round={3}
          timeRemaining={30}
        >
          <Countdown__StyledCountdown
            animationTime={30}
            large={true}
            vb={8}
          >
            <StyledComponent
              animationTime={30}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "Countdown__StyledCountdown-sc-7rbb8p-0",
                    "isStatic": false,
                    "lastClassName": "ljprAI",
                    "rules": Array [
                      "position:relative;margin:auto;height:",
                      [Function],
                      "rem;width:",
                      [Function],
                      "rem;text-align:center;border-radius:100%;box-shadow:0 0 1rem 3px rgba(255,255,255,0.4);.countdown__number{color:white;display:block;line-height:",
                      [Function],
                      "rem;position:absolute;top:0;left:50%;transform:translateX(-50%);font-size:",
                      [Function],
                      "rem;z-index:999;}svg{position:absolute;top:0;right:0;width:",
                      [Function],
                      "rem;height:",
                      [Function],
                      "rem;fill:none;transform:rotateY(-180deg) rotateZ(-90deg);}svg circle{stroke-linecap:butt;stroke-width:4px;stroke:",
                      [Function],
                      ";stroke-dasharray:",
                      [Function],
                      ";stroke-dashoffset:0px;}",
                      [Function],
                      " ",
                      [Function],
                    ],
                  },
                  "displayName": "Countdown__StyledCountdown",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "Countdown__StyledCountdown-sc-7rbb8p-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              large={true}
              vb={8}
            >
              <div
                className="Countdown__StyledCountdown-sc-7rbb8p-0 ljprAI"
              >
                <div
                  className="countdown__number"
                >
                  30
                </div>
                <svg>
                  <circle
                    cx="50%"
                    cy="50%"
                    r={38}
                  />
                </svg>
              </div>
            </StyledComponent>
          </Countdown__StyledCountdown>
        </Countdown>
      </div>
    </StyledComponent>
  </Timer__StyledTimer>
</RoundTimer>
`;

exports[`<FinalJeopardy /> renders final jeopardy form and matches snapshot 1`] = `
<FinalJeopardy
  category={
    Object {
      "__typename": "Category",
      "id": "cat123",
      "name": "animi",
    }
  }
  config={
    Object {
      "__typename": "GameConfig",
      "finalTime": 0,
      "hasDoubleJeopardy": false,
      "hasRoundTimer": false,
      "id": "config123",
      "roundTime": 0,
    }
  }
  gameName="Test Game"
  teams={
    Array [
      Object {
        "__typename": "Team",
        "game": Object {
          "id": "g@m3",
          "name": "Test Game",
        },
        "id": "t1",
        "name": "Team1",
        "score": 1000,
      },
      Object {
        "__typename": "Team",
        "game": Object {
          "id": "g@m3",
          "name": "Test Game",
        },
        "id": "t2",
        "name": "Team2",
        "score": 1000,
      },
    ]
  }
  toggleChosenTeam={[MockFunction]}
>
  <Cover
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "top": "5px",
        "transform": "translateX(-50%)",
        "width": "100%",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Cover-urqkfw-0",
            "isStatic": false,
            "lastClassName": "fCxfBD",
            "rules": Array [
              "position:fixed;top:9.5rem;left:50%;z-index:1000;width:100%;transform-origin:left;box-shadow:0px 0rem 6px 4px rgba(0,0,0,0.6);max-width:",
              [Function],
              ";margin:0 auto;height:calc(100vh - 17rem);background-color:",
              [Function],
              ";color:",
              [Function],
              ";background-image:linear-gradient( 45deg,",
              [Function],
              " 15%,",
              [Function],
              " 50%,",
              [Function],
              " 95%,",
              [Function],
              " );.fullscreen__inner{height:100%;display:grid;grid-template-rows:fit-content(15%) auto fit-content(10%);grid-template-columns:10% 80% 10%;grid-auto-flow:row;}.titles{grid-row:1 / 2;grid-column:2 / 3;align-self:end;&__inner{margin-top:2rem;font-size:1.5em;display:flex;justify-content:space-around;align-items:center;h3{line-height:1.2;color:#ffdb6d;}}}.main__display{align-self:center;grid-row:2 / 3;grid-column:2 / 3;h1{font-size:6vw;}.question,.answer{font-size:4.5vh;padding:1rem 8%;line-height:1.3;letter-spacing:2px;text-transform:uppercase;&--unfocused{padding-bottom:3rem;text-transform:uppercase;line-height:1.1;}&--mounted{font-size:3.5vh;}}.other{font-size:4vh;span{font-weight:bolder;letter-spacing:2px;}}.prompt{font-size:3vh;font-style:italic;color:",
              [Function],
              ";}.answer{color:",
              [Function],
              ";}.congrats{line-height:1.5;letter-spacing:2px;font-size:6vh;}}.controls{grid-row:3 / 4;grid-column:2 / 3;margin-bottom:2rem;font-size:1.5em;&__ctas{display:flex;width:50%;padding:1rem 0;align-items:center;justify-content:space-around;margin:0 auto;}&__message{color:",
              [Function],
              ";font-style:italic;}}",
            ],
          },
          "displayName": "Cover",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Cover-urqkfw-0",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "top": "5px",
          "transform": "translateX(-50%)",
          "width": "100%",
        }
      }
    >
      <ForwardRef
        className="Cover-urqkfw-0 fCxfBD"
        style={
          Object {
            "height": "100%",
            "position": "relative",
            "top": "5px",
            "transform": "translateX(-50%)",
            "width": "100%",
          }
        }
      >
        <div
          className="Cover-urqkfw-0 fCxfBD"
          style={
            Object {
              "height": "100%",
              "position": "relative",
              "top": "5px",
              "transform": "translateX(-50%)",
              "width": "100%",
            }
          }
        >
          <div
            className="fullscreen__inner"
          >
            <div
              className="titles"
            >
              <div
                className="titles__inner"
              >
                <h3>
                  animi
                </h3>
              </div>
              <Utilities__Divider>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Utilities__Divider-s11g5i-5",
                        "isStatic": true,
                        "lastClassName": "jrubw",
                        "rules": Array [
                          "width:80%;opacity:0.5;margin:1rem auto;",
                        ],
                      },
                      "displayName": "Utilities__Divider",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Utilities__Divider-s11g5i-5",
                      "target": "hr",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <hr
                    className="Utilities__Divider-s11g5i-5 jrubw"
                  />
                </StyledComponent>
              </Utilities__Divider>
            </div>
            <TeamWagers
              nextStage={[Function]}
              setWagers={[Function]}
              teams={
                Array [
                  Object {
                    "__typename": "Team",
                    "game": Object {
                      "id": "g@m3",
                      "name": "Test Game",
                    },
                    "id": "t1",
                    "name": "Team1",
                    "score": 1000,
                  },
                  Object {
                    "__typename": "Team",
                    "game": Object {
                      "id": "g@m3",
                      "name": "Test Game",
                    },
                    "id": "t2",
                    "name": "Team2",
                    "score": 1000,
                  },
                ]
              }
            >
              <div
                className="main__display"
              >
                <Wagers__WagerInputs>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Wagers__WagerInputs-sc-1tb6h8w-0",
                          "isStatic": false,
                          "lastClassName": "bKaJba",
                          "rules": Array [
                            "background-color:white;color:black;padding:3rem;font-size:2rem;min-width:50rem;max-width:60rem;margin:0 auto;label{padding:1rem 20%;display:flex;justify-content:space-between;align-items:center;span{font-size:0.8em;color:",
                            [Function],
                            ";}}input{padding:1rem;width:8rem;font-size:1.5rem;color:",
                            [Function],
                            ";&:focus{outline-color:",
                            [Function],
                            ";}}.disclaimer{color:",
                            [Function],
                            ";font-size:italic;font-size:0.9em;margin-top:1rem;&.smaller{font-size:0.8em;}}",
                          ],
                        },
                        "displayName": "Wagers__WagerInputs",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Wagers__WagerInputs-sc-1tb6h8w-0",
                        "target": "fieldset",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <fieldset
                      className="Wagers__WagerInputs-sc-1tb6h8w-0 bKaJba"
                    >
                      <Utilities__TextEmph>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "Utilities__TextEmph-s11g5i-0",
                                "isStatic": false,
                                "lastClassName": "jQVwwM",
                                "rules": Array [
                                  "font-weight:bolder;color:",
                                  [Function],
                                  ";font-size:1.3em;",
                                ],
                              },
                              "displayName": "Utilities__TextEmph",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "Utilities__TextEmph-s11g5i-0",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="Utilities__TextEmph-s11g5i-0 jQVwwM"
                          >
                            Choose Final Wagers
                          </span>
                        </StyledComponent>
                      </Utilities__TextEmph>
                      <label
                        htmlFor="wager0"
                        key="wager-0"
                      >
                        Team1
                        <span>
                          $
                           
                          <input
                            max={1000}
                            min={0}
                            name="wager0"
                            onChange={[Function]}
                            onKeyUp={[Function]}
                            required={true}
                            step={100}
                            type="number"
                            value={0}
                          />
                        </span>
                      </label>
                      <label
                        htmlFor="wager1"
                        key="wager-1"
                      >
                        Team2
                        <span>
                          $
                           
                          <input
                            max={1000}
                            min={0}
                            name="wager1"
                            onChange={[Function]}
                            onKeyUp={[Function]}
                            required={true}
                            step={100}
                            type="number"
                            value={0}
                          />
                        </span>
                      </label>
                      <p
                        className="disclaimer"
                      >
                        You may have to write down your team
                        '
                        s answer
                      </p>
                      <p
                        className="disclaimer smaller"
                      >
                        ...And cover your eyes for the other teams wager 🙈
                      </p>
                    </fieldset>
                  </StyledComponent>
                </Wagers__WagerInputs>
              </div>
              <div
                className="controls"
              >
                <Utilities__Divider>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Utilities__Divider-s11g5i-5",
                          "isStatic": true,
                          "lastClassName": "jrubw",
                          "rules": Array [
                            "width:80%;opacity:0.5;margin:1rem auto;",
                          ],
                        },
                        "displayName": "Utilities__Divider",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Utilities__Divider-s11g5i-5",
                        "target": "hr",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <hr
                      className="Utilities__Divider-s11g5i-5 jrubw"
                    />
                  </StyledComponent>
                </Utilities__Divider>
                <Button
                  game={true}
                  onClick={[Function]}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Button-sc-1f1w3gq-0",
                          "isStatic": false,
                          "lastClassName": "gfKOtE",
                          "rules": Array [
                            "font-family:",
                            [Function],
                            ";color:white;border-radius:0.5rem;text-transform:uppercase;letter-spacing:1px;transition:.2s ease;&:hover{cursor:pointer;}&:focus{cursor:pointer;outline:",
                            [Function],
                            " auto 3px;}",
                            [Function],
                            " ",
                            [Function],
                            " ",
                            [Function],
                            " ",
                            [Function],
                            " ",
                            [Function],
                            " &.button--sound{font-weight:bolder;text-decoration:line-through;&::after{content:\\"❌\\";padding-left:3px;}",
                            [Function],
                            "}",
                            [Function],
                          ],
                        },
                        "displayName": "Button",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Button-sc-1f1w3gq-0",
                        "target": "button",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    game={true}
                    onClick={[Function]}
                  >
                    <button
                      className="Button-sc-1f1w3gq-0 gfKOtE"
                      onClick={[Function]}
                    >
                      Start Final Round
                    </button>
                  </StyledComponent>
                </Button>
              </div>
            </TeamWagers>
          </div>
        </div>
      </ForwardRef>
    </StyledComponent>
  </Cover>
</FinalJeopardy>
`;
